rust/otap-dataflow/crates/otap/src/experimental/gigla_exporter/transformer.rs

```diff
@@ -0,0 +1,684 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use opentelemetry_proto::tonic::collector::logs::v1::ExportLogsServiceRequest;
+use opentelemetry_proto::tonic::common::v1::any_value::Value as OtelAnyValueEnum;
+use serde_json::{Value, json};
+
+use super::config::{Config, SchemaConfig};
+
+const ATTRIBUTES_FIELD: &str = "attributes";
+
+// TODO: Performance review and improvements
+/// Converts OTLP logs to Azure Log Analytics format
+pub struct Transformer {
+    schema: SchemaConfig,
+}
+
+impl Transformer {
+    /// Create a new transformer with the given configuration.
+    /// Must be used; constructing and discarding would be a no-op.
+    #[must_use]
+    pub fn new(config: &Config) -> Self {
+        Self {
+            schema: config.api.schema.clone(),
+        }
+    }
+
+    /// Convert OTLP logs to flat JSON objects for Log Analytics.
+    /// Must be used; otherwise transformed entries are lost.
+    #[must_use]
+    // TODO: Remove print_stdout after logging is set up
+    #[allow(clippy::print_stdout)]
+    pub fn convert_to_log_analytics(&self, request: &ExportLogsServiceRequest) -> Vec<Value> {
+        let mut entries = Vec::new();
+
+        for resource_logs in &request.resource_logs {
+            let resource_attrs = if !self.schema.disable_schema_mapping {
+                // Schema mapping enabled: only add mapped resource attributes
+                self.apply_resource_mapping(&resource_logs.resource)
+            } else {
+                // Schema mapping disabled: no resource attributes in legacy format
+                serde_json::Map::new()
+            };
+
+            for scope_logs in &resource_logs.scope_logs {
+                let scope_attrs = if !self.schema.disable_schema_mapping {
+                    // Schema mapping enabled: only add mapped scope attributes
+                    self.apply_scope_mapping(&scope_logs.scope)
+                } else {
+                    // Schema mapping disabled: no scope attributes in legacy format
+                    serde_json::Map::new()
+                };
+
+                for log_record in &scope_logs.log_records {
+                    let mut entry = serde_json::Map::new();
+
+                    if self.schema.disable_schema_mapping {
+                        // Legacy transform when schema mapping is disabled
+                        Self::legacy_transform(&mut entry, log_record);
+                    } else {
+                        // Apply configured mappings when schema mapping is enabled
```

Comment:
nit - remove this comment, as mapping is already applied in earlier steps. We are only inserting the mapped schema attributes here.